isacc_messaging/api/views.py

```diff
@@ -88,22 +92,43 @@ def auditlog_addevent():
 
 
 @base_blueprint.route("/MessageStatus", methods=['POST'])
-def message_status_update():
+def message_status_update(callback_req=None, attempt_count=0):
+    """Registered callback for Twilio to transmit updates
+
+    As Twilio occasionally hits this callback prior to local data being
+    available, it is also called subsequently from a job queue.  The
+    parameters are only defined in the retry state.
+
+    :param req: request from a job queue
+    :param attempt_count: the number of failed attemts thus far, only
+        defined from job queue
+    """
+    use_request = request
+    if callback_req:
+        use_request = callback_req
+
     audit_entry(
         f"Call to /MessageStatus webhook",
-        extra={'request.values': dict(request.values)},
+        extra={'use_request.values': dict(use_request.values)},
         level='debug'
     )
 
     record_creator = IsaccRecordCreator()
     try:
-        record_creator.on_twilio_message_status_update(request.values)
+        record_creator.on_twilio_message_status_update(use_request.values)
     except Exception as ex:
         audit_entry(
             f"on_twilio_message_status_update generated error {ex}",
             level='error'
         )
-        return ex, 200
+        # Couldn't locate the message, most likely means twilio was quicker
+        # to call back, than HAPI could persist and find.  Push to REDIS
```

Comment:
is there a way to add a check (or delay) after persisting something in Hapi, to ensure it's actually searchable?

seems a little more robust to fail fast ie ensure we're ready to accept twilio callbacks than than keep trying in a loop

Reply:
the two are asynchronous events.  we don't receive the necessary SID (stored as an identifier in the Communication) until we ask twilio to send the message.
this is a different thread of execution, where we're now looking up the value that came in from the twilio status update callback.

yes, we could add another round trip to HAPI, to confirm the identifier we added can be found, and raise an error if not, but that isn't going to prevent twilio from hitting our callback, as the request was already sent to them.